python/lsst/ip/isr/crosstalk.py

```diff
@@ -167,255 +600,53 @@ def run(self, exposure, crosstalkSources=None, isTrimmed=False):
+        if not crosstalk:
+            crosstalk = CrosstalkCalib(log=self.log)
+            crosstalk = crosstalk.fromDetector(exposure.getDetector(),
+                                               coeffVector=self.config.crosstalkValues)
+            # detector = exposure.getDetector()
+            # if not self.config.hasCrosstalk(detector=detector):
+            # raise RuntimeError("Attempted to correct crosstalk without crosstalk coefficients")
+            # coeffs = self.config.getCrosstalk(detector=detector)
+        if not crosstalk.log:
+            crosstalk.log = self.log
+        if not crosstalk.hasCrosstalk:
+            raise RuntimeError("Attempted to correct crosstalk without crosstalk coefficients.")
+            self.log.warn("Attempted to correct crosstalk without crosstalk coefficients.")
         else:
-            detector = exposure.getDetector()
-            if not self.config.hasCrosstalk(detector=detector):
-                raise RuntimeError("Attempted to correct crosstalk without crosstalk coefficients")
-            coeffs = self.config.getCrosstalk(detector=detector)
-
-        self.log.info("Applying crosstalk correction.")
-        subtractCrosstalk(exposure, crosstalkCoeffs=coeffs,
-                          minPixelToMask=self.config.minPixelToMask,
-                          crosstalkStr=self.config.crosstalkMaskPlane, isTrimmed=isTrimmed,
-                          backgroundMethod=self.config.crosstalkBackgroundMethod)
-
-
-# Flips required to get the corner to the lower-left
-# (an arbitrary choice; flips are relative, so the choice of reference here is not important)
-X_FLIP = {lsst.afw.cameraGeom.ReadoutCorner.LL: False, lsst.afw.cameraGeom.ReadoutCorner.LR: True,
-          lsst.afw.cameraGeom.ReadoutCorner.UL: False, lsst.afw.cameraGeom.ReadoutCorner.UR: True}
-Y_FLIP = {lsst.afw.cameraGeom.ReadoutCorner.LL: False, lsst.afw.cameraGeom.ReadoutCorner.LR: False,
-          lsst.afw.cameraGeom.ReadoutCorner.UL: True, lsst.afw.cameraGeom.ReadoutCorner.UR: True}
+            self.log.info("Applying crosstalk correction.")
+            crosstalk.subtractCrosstalk(exposure, crosstalkCoeffs=crosstalk.coeffs,
+                                        minPixelToMask=self.config.minPixelToMask,
+                                        crosstalkStr=self.config.crosstalkMaskPlane, isTrimmed=isTrimmed,
+                                        backgroundMethod=self.config.crosstalkBackgroundMethod)
+
+            if crosstalk.interChip:
+                if crosstalkSources:
+                    for detName in crosstalk.interChip:
+                        if isinstance(crosstalkSources[0], 'lsst.afw.image.Exposure'):
+                            # Received a proper exposure
+                            sourceNames = [exp.getDetector().getName() for exp in crosstalkSources]
+                        else:
+                            # Received deferred dataset
```

Comment:
Not immediately clear what "deferred dataset" means here

Reply:
Switched this and previous comment to use abbreviated python class to clarify.  This now points to dafButler.DeferredDatasetHandle.